js/client-mainmenu.js

```diff
@@ -664,9 +664,7 @@
 		openChallenge: function (name, $pmWindow) {
 			if (!$pmWindow) $pmWindow = this.openPM(name, true);
 			var $challenge = $pmWindow.find('.challenge');
-			if (!$challenge.length) {
-				$challenge = $('<div class="challenge"></div>').insertAfter($pmWindow.find('h3'));
-			}
+			$challenge = $('<div class="challenge"></div>').insertAfter($challenge.length ? $challenge : $pmWindow.find('h3'));
```

Comment:
Err, the window having multiple challenges is not a supported state...

Reply:
The code assumes the challenge window a single window that can be found with `$pmWindow.find('.challenge')`.

A lot of code will break if that assumption is false, such as:

https://github.com/Zarel/Pokemon-Showdown-Client/blob/89a3ec78c0c3d6128135f73919c66ab768490862/js/client-mainmenu.js#L641-L658